src/pointer-analysis/goto_program_dereference.h

```diff
@@ -19,6 +19,8 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "value_sets.h"
 #include "value_set_dereference.h"
 
+class guardt;
```

Comment:
I'm not exactly sure about the order of the commits here, but at the very end this shouldn't be needed anymore at all I think?

Reply:
It is still used in some unused deprecated function and I'm not sure of what to do with all these functions (most of the code in this class is unused).

Reply:
Remove the cruft. `git revert` is not all that hard.

Reply:
I added a commit to replace that argument by an exprt. I also updated the date of commits so that they appear in the right order.